File: storage/data_original/at_coder/abc023/B/4806479.py
# ??
N=int(input())
S=input()

# ??
c=0

# ???
x='b'

# ????
if x==S:
  print(c)
  exit()

# ????? 1->2->3->1->...
f=1

# ??100???
for i in range(100):
  # ?????1???????'a'????'c'?
  if f==1:
    x='a'+x+'c'
    f=2
  # ?????2???????'c'????'a'?
  elif f==2:
    x='c'+x+'a'
    f=3
  # ?????3???????'b'????'b'?
  elif f==3:
    x='b'+x+'b'
    f=1
  # ????????
  c+=1
  # ????
  if x==S:
    print(c)
    exit()

# ??????????'-1'???
print(-1)

File: data/data_transpiled/at_coder/abc023/B/4806479.java
public static int N = Integer . parseInt ( input ) {
  String S = input . nextLine ( ) ;
  int c = 0 ;
  String x = "b" ;
  if ( x . equals ( S ) ) {
    System . out . println ( c ) ;
    exit ( ) ;
  }
  int f = 1 ;
  for ( int i = 0 ;
  i < 100 ;
  i ++ ) {
    switch ( f ) {
      case 1 : x = "a" + x + "c" ;
      f = 2 ;
      break ;
      case 2 : x = "c" + x + "a" ;
      f = 3 ;
      break ;
      case 3 : x = "b" + x + "b" ;
      f = 1 ;
    }
    c ++ ;
    if ( x . equals ( S ) ) {
      System . out . println ( c ) ;
      exit ( ) ;
    }
  }
  System . out . println ( - 1 ) ;
  return c ;
}
